### PR TITLE
Add trimws

### LINKED ISSUE #411: Add trimws()
Introduced in R 3.2.0, see [src/library/base/R/strwrap.R#219](https://github.com/wch/r-source/blob/5a156a0865362bb8381dcd69ac335f5174a4f60c/src/library/base/R/strwrap.R#L219).

This function is also used by `read.dcv()`, see [src/library/base/R/dcf.R#126](https://github.com/wch/r-source/blob/5a156a0865362bb8381dcd69ac335f5174a4f60c/src/library/base/R/dcf.R#L126).

### PR DESCRIPTION
**Pull request by strboul**
Fixes #411

### CODE REVIEW

File: core/src/main/R/base/strwrap.R [hunk 1]
```diff
@@ -191,3 +191,16 @@ function(x, y, style = c("table", "list"),
     }
     r
 }
+
+trimws <- function(x, which = c("both", "left", "right"),
+                   whitespace = "[ \t\r\n]")
+{
+    which <- match.arg(which)
+    mysub <- function(re, x) sub(re, "", x, perl = TRUE)
```
Comment (akbertram) [inline core/src/main/R/base/strwrap.R]: Avoid definining an inline function here. That causes problems (currently) for Renjin's optimizer, and makes the code harder to read. You can just make the switch statement an argument of the sub function as the pattern is the only thing the changes.

File: core/src/main/R/base/strwrap.R [hunk 2]
```diff
@@ -191,3 +191,16 @@ function(x, y, style = c("table", "list"),
     }
     r
 }
+
+trimws <- function(x, which = c("both", "left", "right"),
+                   whitespace = "[ \t\r\n]")
```
Comment (akbertram) [inline core/src/main/R/base/strwrap.R]: Is there a `whitespace` argument defined in GNU R? I'd prefer to keep the base function API exactly the same.

Comment (strboul) [reply core/src/main/R/base/strwrap.R]: Yes. The function is just mirrored from the latest R version (`3.5.2`).

File: core/src/main/R/base/strwrap.R [hunk 3]
```diff
@@ -196,7 +196,9 @@ trimws <- function(x, which = c("both", "left", "right"),
                    whitespace = "[ \t\r\n]")
 {
     which <- match.arg(which)
-    mysub <- function(re, x) sub(re, "", x, perl = TRUE)
+    mysub <- function(re, x) {
```
Comment (akbertram) [inline core/src/main/R/base/strwrap.R]: By "inline function", I meant a function defined in another function. But that's ok, the optimizer needs to handle inline functions at some point anyway.

Comment (strboul) [reply core/src/main/R/base/strwrap.R]: I think I misread it as "one line function."

**Pull request merged**